poleno/utils/tests/test_templatetags.py

```diff
@@ -253,3 +256,9 @@ def test_change_lang_tag(self):
             self.assertEqual(r1.content, u'(/en/language/)(/de/language/)(/fr/language/)')
             self.assertEqual(r2.content, u'(/en/language/)(/de/language/)(/fr/language/)')
             self.assertEqual(r3.content, u'(/en/language/)(/de/language/)(/fr/language/)')
+
```

Comment:
@viliambalaz Nepotrebujeme `with self.settings(LANGUAGES=lang):` aj v tomto teste?

Reply:
@martinmacko47 Funguje to s aj bez toho. Zakazdym sa redirectne na `/en/languages/`

Reply:
@martinmacko47 K comu sa vstahuje komentar "Fix active languages" `with self.settings(LANGUAGES=lang): # Fix active languages`

Reply:
@viliambalaz V settingoch v CHV aplikacii mame

```
LANGUAGES = (
    (u'sk', u'Slovensky'),
    (u'en', u'English'),
    )
```

Ale kedze `poleno/utils` su vseobecna libka, jej testy by sa nemali spoliehat na settingy v aplikacii. Preto je lepsie v teste explicitne povedat, nad akou hodnotou `LANGUAGES` test bezi. Aby ked napriklad v buducnosti (alebo na produkcii) zmenime setting `LANGUAGES` iba na jeden jazyk, tak aby test stale fungoval.

Takze treba setting overridnut bud s tym ako mame v predoslom teste. Ale kludne aj cokolvek ine, co ma aspon 2 jazyky.

```
lang = ((u'de', u'Deutsch'), (u'en', u'English'), (u'fr', u'Francais'))
```

Co sa tyka komentaru `# Fix active languages`, asi je tam to slovo "Fix" matuce. Nie je to vo vyzname "opravit active languages". Ale vo vyzname "zafixovat active languages". Kludne ten komentar vyhod alebo prepis ak myli.

Reply:
@martinmacko47 Som nato siel asi zle. Nato aby som otestoval novu funkcionalitu template tagu `change_lang` potrebujem ostranit `django.middleware.locale.LocaleMiddleware`. Tym padom viem tento test napisat takto:
```
        with self.settings(
                MIDDLEWARE_CLASSES=[mc for mc in settings.MIDDLEWARE_CLASSES
                                    if mc != u'django.middleware.locale.LocaleMiddleware']):
            r = self.client.get(u'/language/')
            self.assertIs(type(r), HttpResponseNotFound)
            self.assertEqual(r.content, u'(/language/)')
```
Takze return hodnota `HttpRepsonseNotFound` z `page_not_found` sa uz dalej nebude spracovavat. Pricom sa neberie ohlad na definovane jazyky `LANGUAGES`.

Ak `LocaleMiddleware` je zadefinovany, potom nastane `HttpResponseRedirect`, pricom language sa zvoli z `LANGUAGE_CODE`, ktory mame nastaveny v `CustomTestRunner` na `'en'`.
Mam napisat testy s/bez LocaleMiddleware? Podla mna staci otestovat iba nsu funkcionalitu, teda bez LocalMiddleware.

Reply:
@viliambalaz Ano, asi mas pravdu. Takto to bude asi lepsie, kedze sa bude dat explicitne overit, co `change_lang` vygeneruje. Skus ten test teda takto prepisat.

Kazdopadne, aj tak budes potrebovat overridnut aj `LANGUAGES` setting, ako v teste o jednovyssie, aby si mal istotu, ake languages mas nastave.

Reply:
:ok: